lxd/instance/drivers/driver_qemu.go

```diff
@@ -7128,8 +7159,9 @@ func (d *qemu) MigrateReceive(args instance.MigrateReceiveArgs) error {
 			}
 
 			for _, name := range offerHeader.SnapshotNames {
+				snapshotName := name
 				base := instance.SnapshotToProtobuf(apiInstSnap)
-				base.Name = &name
+				base.Name = &snapshotName
```

Comment:
Does this fix a bug? Can we have this as a separate commit please

Reply:
This is also suggested by the linter : `Implicit memory aliasing in for loop.`. But I think this can be avoided like this:
```go
for i := range offerHeader.SnapshotNames {
  ...
  base.Name = &offerHeader.SnapshotNames[i]
  ...
}
```

Reply:
Yeah i can see what the issue is, im surprised its not caused an issue before. I like your suggested fix here ^

Reply:
But I think its worth checking why this wasn't causing issues before, it seems like multiple snapshots would all end up pointing to the same name.

Reply:
I tried to live migrate an instance (backed by shared storage) with two snapshots (`snap0` and `snap1`). The migration works fine. The snapshots are still named `snap0` and `snap1` after the migration. Could this be because the data is not moving around because of the shared nature of thee storage? I'll try with local storage..